### PR TITLE
repr: use postgres's float input and output formats

### LINKED ISSUE #1533: Testdrive: correct support for jsonb
I _think_ that to handle this correctly we need to make testdrive lightly parse individual rows so that the json strings in each row don't get split up by whitespace.

Better diffs (sub-line) would also be nice, but aren't a requirement.

### LINKED ISSUE #1553: Cast 'NaN' to float
```
jamie=> select 'Inf'::float;
 ?column? 
----------
      inf
(1 row)

jamie=> select '-Inf'::float;
 ?column? 
----------
     -inf
(1 row)

jamie=> select 'NaN'::float;
 ?column? 
----------
         
(1 row)
```

The culprit is probably the `.to_lowercase` in parse_float64.

### PR DESCRIPTION
**Pull request by benesch**
@sploiselle certainly feel free to review the whole thing, but cc'd you for the docs review!

Fix #1553.

### CODE REVIEW

File: doc/user/sql/types/float.md [hunk 1]
```diff
@@ -22,7 +22,29 @@ Type               | Aliases           | Size    | Range
 
 - Materialize assumes untyped numeric literals containing decimal points are [`decimal`](../decimal); to use `float`, you must explicitly cast them as we've done below.
 
-## Details
```
Comment (sploiselle) [inline doc/user/sql/types/float.md]: Just for consistency with other docs, can you leave this h2 in place?

File: doc/user/sql/types/float.md [hunk 2]
```diff
@@ -22,7 +22,29 @@ Type               | Aliases           | Size    | Range
...
+### Special values
+
+Floating-point numbers have three special values, as specified in IEEE 754:
+
+Value       | Aliases                    | Represents
+------------|----------------------------|-----------
+`NaN`       |                            | not a number
```
Comment (sploiselle) [inline doc/user/sql/types/float.md]: I prefer capitalization of each `td` element that doesn't being with `code`

File: doc/user/sql/types/float.md (hunk 1 above)
Comment (benesch) [reply doc/user/sql/types/float.md]: There are two `## Details` headers in this particular file. Seemed like the first one was the right one to keep?

File: doc/user/sql/types/float.md (hunk 2 above)
Comment (benesch) [reply doc/user/sql/types/float.md]: Sure, done!

File: doc/user/sql/types/float.md (hunk 1 above)
Comment (sploiselle) [reply doc/user/sql/types/float.md]: Ah, you're right! I just looked at the diff and saw it got yanked. Apologies.

**Pull request merged**